app/Services/DependencyService.php

```diff
@@ -20,28 +54,43 @@ public static function touchDependencyByKey(string $key, ?User $user = null): vo
         }
 
         if ($dependency->per_user && $user === null) {
-            Log::error(sprintf('Dependency %s is per user but user was not specifieid', $key));
+            Log::error(sprintf('Dependency %s is per user but user was not specified', $key));
             return;
         }
 
         if ($dependency->per_user) {
-            self::touchUserDependency($dependency, $user);
+            self::touchUserDependency($dependency);
         } else {
             self::touchGlobalDependency($dependency);
         }
     }
 
+    private static function removeDependencyFromCache(Dependency $dependency)
+    {
+        Cache::forget(self::getDependencyCacheKey($dependency));
+    }
+
+    /**
+     * Update the time on the dependency and re-cache it
+     * @param Dependency $dependency
+     */
     public static function touchGlobalDependency(Dependency $dependency): void
     {
         $dependency->touch();
+        self::removeDependencyFromCache($dependency);
     }
 
-    public static function touchUserDependency(Dependency $dependency, User $user): void
+    public static function touchUserDependency(Dependency $dependency): void
     {
+        if (($user = Auth::user()) === null) {
```

Comment:
Can you not check this via middleware rather than having to do it in the controller?

Reply:
This is more a logic thing, rather than an authentication thing - we're checking auth in the auth middleware still :)